src/components/block/Block.ts

```diff
@@ -0,0 +1,278 @@
+    // Используйте шаблонизатор из npm или напишите свой безопасный
+    // Нужно не в строку компилировать (или делать это правильно),
+    // либо сразу в DOM-элементы возвращать из compile DOM-ноду
+  }
+
+  // Может переопределять пользователь, необязательно трогать
+  render(): DocumentFragment {
+    return new DocumentFragment();
+  }
+
+  getContent() {
+    return this._element;
+  }
+
+  compile(template: string, props: Props) {
+    const propsWithStubs: Props = {...props};
+
+    Object.entries(this.children).forEach(([key, child]) => {
+      if (Array.isArray(child)) {
+        propsWithStubs[key] = child
+          .map((childItem) => `<div data-id="${childItem._id}"></div>`)
+          .join(' ');
+      } else {
+        propsWithStubs[key] = `<div data-id="${child._id}"></div>`;
+      }
+    });
+
+    const fragment = this._createDocumentElement('template') as HTMLTemplateElement;
+
+    fragment.innerHTML = new Templator(template).compile(propsWithStubs);
+
+    Object.values(this.children).forEach((child) => {
+      if (Array.isArray(child)) {
+        child.forEach((childItem) => {
+          const stub = fragment.content.querySelector(`[data-id="${childItem._id}"]`);
+          stub?.replaceWith(childItem.getContent() as Node);
+        });
+      } else {
+        const stub = fragment.content.querySelector(`[data-id="${child._id}"]`);
+        stub?.replaceWith(child.getContent() as Node);
+      }
+    });
+
+    return fragment.content;
+  }
+
+  _makePropsProxy(props: Props) {
+    // Можно и так передать this
+    // Такой способ больше не применяется с приходом ES6+
+    // const self = this;
+
+    const eventBus = this.eventBus.bind(this);
+
+    return new Proxy(props, {
+      set(target, prop: string, value) {
+        // this здесь указывает на объект handler
+        const _target = getDeepCopy(props);
+        if (prop in target) {
+          target[prop] = value;
+          eventBus().emit(Block.EVENTS.FLOW_CDU, _target, props);
+          return true;
+        }
+        return false;
+      },
+    });
+  }
+
+  _createDocumentElement(tagName: string) {
+    // Можно сделать метод, который через фрагменты в цикле создаёт сразу несколько блоков
+    const element = document.createElement(tagName);
+    element.setAttribute('data-id', this._id ?? '');
+    return element;
+  }
+
+  show() {
+    const content = this.getContent();
+    if (content) {
+      content.style.display = 'block';
+    }
+  }
+
+  hide() {
+    const content = this.getContent();
+    if (content) {
+      content.style.display = 'none';
+    }
+  }
+
+  _addEvents() {
+    const {events} = this.props;
+
+    if (events) {
+      Object.keys(events).forEach((eventName) => {
+        this._element?.addEventListener(
+          eventName,
+          events[eventName].event,
+          events[eventName].useCapture ?? false,
+        );
+      });
+    }
+  }
+
+  _removeEvents() {
+    const events = {...this.props.events};
```

Comment:
Функции передаются по ссылке. Нет смысла деструктуризовать их из одного объекта в другой
```suggestion
    const {events} = this.props;
```